src/librustc_lsan/lib.rs

```diff
@@ -17,4 +17,5 @@
             reason = "internal implementation detail of sanitizers",
             issue = "0")]
 
+#[allow(unused_extern_crates)]
```

Comment:
I take it from this that you removed the whitelist? I don't think we should be warning about allocator crates, since we *know* that they server a purpose, right? 

Before, you were asking for some tips on how to detect allocator crates. Ordinarily, if we knew the `CrateNum` of the crate in question, let's call it `cnum`, then we could do this by calling `tcx.is_allocator(cnum.as_def_id())` (where `tcx` is the "type context"). However, from what I can tell, this lint is generated during name resolution, at which point the `tcx` doesn't exist yet -- moreover I don't see an obvious way to extract the crate number. 

So I'm not sure the best way to fix this just now.